src/types.jl

```diff
@@ -0,0 +1,68 @@
+abstract type AbstractGeometry end
+"""The overcoupling `Geometry` type."""
+struct Geometry <: AbstractGeometry end
+
+abstract type AbstractPoint <: AbstractGeometry end
+"""A simple `Point`."""
+struct Point <: AbstractPoint end
+
+abstract type AbstractCurve <: AbstractGeometry end
+abstract type AbstractLineString <: AbstractCurve end
+"""A `LineString` is a collection of straight lines between its `Point`s."""
+struct LineString <: AbstractCurve end
+"""A Line is `LineString` with just two points."""
+struct Line <: LineString end
```

Comment:
This is invalid, you can only subtype abstract structs. If these relations are desired, perhaps they all should be abstract.